src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.DocumentChangeTracker.cs

```diff
@@ -59,6 +60,20 @@ public ImmutableArray<SourceText> GetTrackedTexts()
             public LspWorkspaceManager GetLspWorkspaceManager() => _queue._lspWorkspaceManager;
 
             public bool IsComplete() => _queue._queue.IsCompleted && _queue._queue.IsEmpty;
+
+            public async Task<bool> AreAllItemsCancelledAsync()
+            {
+                while (!_queue._queue.IsEmpty)
+                {
+                    var item = await _queue._queue.DequeueAsync().ConfigureAwait(false);
+                    if (!item.CombinedCancellationToken.IsCancellationRequested)
```

Comment:
if this is false, then the item was not canceled, but it was dequeued from teh queue... seems really strange.  doc heavily if this is actualy what we want.

Reply:
didn't quite get this one actually - but this was changed to `cancellationToken.ThrowIfCancellationRequested();`  This just exits early if the work item was cancelled while in queue so we don't even try to run it.